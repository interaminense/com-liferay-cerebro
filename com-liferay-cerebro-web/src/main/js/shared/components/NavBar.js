import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

const DISPLAYS = ['dark', 'light'];

class Brand extends Component {
	render() {
		const {children, href} = this.props;

		return do {
			if (href) {
				<a {...this.otherProps()} class="navbar-brand" href={href}>
					{children}
				</a>;
			} else {
				<div {...this.otherProps()} class="navbar-brand">
					{children}
				</div>;
			}
		};
	}
}

Brand.PROPS = {
	href: Config.string()
};

class NavBar extends Component {
	getChildContext() {
		return {navBar: true};
	}

	render() {
		const {children, display, expand} = this.props;

		const classes = getCN('navbar', {
			'navbar-dark': display === 'dark',
			'navbar-expand': expand,
			'navbar-light': display === 'light'
		});

		return <nav class={classes}>{children}</nav>;
	}
}

NavBar.PROPS = {
	display: Config.oneOf(DISPLAYS),
	expand: Config.bool().value(false)
};

NavBar.Brand = Brand;

export default NavBar;