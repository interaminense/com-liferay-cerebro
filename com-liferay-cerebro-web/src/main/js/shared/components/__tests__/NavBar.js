import NavBar from '../NavBar';

describe('NavBar', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new NavBar();

		expect(component).toMatchSnapshot();
	});

	it('should render Brand as a child', () => {
		component = new NavBar({
			children: [<NavBar.Brand key="foo">{'foo bar'}</NavBar.Brand>]
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with light display', () => {
		component = new NavBar({display: 'light'});

		expect(component).toMatchSnapshot();
	});

	it('should render with dark display', () => {
		component = new NavBar({display: 'dark'});

		expect(component).toMatchSnapshot();
	});
});