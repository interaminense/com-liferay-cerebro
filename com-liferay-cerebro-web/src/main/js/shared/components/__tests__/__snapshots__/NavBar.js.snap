// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render 1`] = `<nav class="navbar"></nav>`;

exports[`NavBar should render Brand as a child 1`] = `
<nav class="navbar">
  <div class="navbar-brand">foo bar</div>
</nav>
`;

exports[`NavBar should render with dark display 1`] = `<nav class="navbar navbar-dark"></nav>`;

exports[`NavBar should render with light display 1`] = `<nav class="navbar navbar-light"></nav>`;
